build.gradle

```diff
@@ -378,15 +380,34 @@ dependencies {
     compileOnly "org.opensearch.plugin:opensearch-scripting-painless-spi:${opensearch_version}"
     compileOnly "org.opensearch:opensearch-job-scheduler-spi:${job_scheduler_version}"
     compile "org.opensearch:common-utils:${common_utils_version}"
+    compile "org.opensearch.client:opensearch-rest-client:${opensearch_version}"
     compile group: 'com.google.guava', name: 'guava', version:'29.0-jre'
     compile group: 'org.apache.commons', name: 'commons-math3', version: '3.6.1'
     compile group: 'com.google.code.gson', name: 'gson', version: '2.8.6'
     compile group: 'com.yahoo.datasketches', name: 'sketches-core', version: '0.13.4'
     compile group: 'com.yahoo.datasketches', name: 'memory', version: '0.12.2'
     compile group: 'commons-lang', name: 'commons-lang', version: '2.6'
-    compile 'software.amazon.randomcutforest:randomcutforest-core:1.0'
-    compile 'software.amazon.randomcutforest:randomcutforest-serialization-json:1.0'
-    compile "org.opensearch.client:opensearch-rest-client:${opensearch_version}"
+    compile group: 'org.apache.commons', name: 'commons-pool2', version: '2.10.0'
+
+    // randomcutforest-serialization uses jackson 2.12, but opensearch-scripting-painless-spi uses jackson 2.11.
+    // compile scope won't work due to conflict.
+    // resolutionStrategy using 2.11 won't work as 
+    // com.fasterxml.jackson.databind.ObjectMapper depends on com/fasterxml/jackson/core/util/JacksonFeature
+    // that is created since 2.12. Compile won't fail but there is a runtime ClassNotFoundException
+    // due to absent JacksonFeature.
+    // The fix is to put jackson in direct dependency and use implementation scope.
+    // implementation scope let the dependency in both compiling and running classpath, but
+    // not leaked through to clients (Opensearch). Here we force the jackson version to whatever
+    // opensearch uses.
+    compile 'software.amazon.randomcutforest:randomcutforest-core:2.0-rc2'
+    implementation 'software.amazon.randomcutforest:randomcutforest-serialization:2.0-rc2'
+    implementation "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
+    implementation "com.fasterxml.jackson.core:jackson-databind:${versions.jackson}"
+    implementation "com.fasterxml.jackson.core:jackson-annotations:${versions.jackson}"
+
+    // used for serializing/deserializing rcf models.
+    compile group: 'io.protostuff', name: 'protostuff-core', version: '1.7.2'
```

Comment:
Why don't use the latest version 1.7.4?

Reply:
Josh/Sudipto did experiments on 1.7.2: https://github.com/aws/random-cut-forest-by-aws/blob/8ad06c93f6944f32b9fa7afb41c16c226da1a003/Java/benchmark/pom.xml#L63 So I am following them.  The versions are always changing.  I will change when there is is feature we want to use in the future.

Reply:
Got it. From https://github.com/protostuff/protostuff/blob/master/CHANGELOG.md, the 1.7.3 fixed one bug, not sure if they have backported this fix to 1.7.2.

Reply:
will change to 1.7.4 and retest then.